core/thread/test/testRWLock.cxx

```diff
@@ -0,0 +1,398 @@
+// #include "TReentrantRWLock.h"
+#include "TVirtualMutex.h"
+#include "TMutex.h"
+#include "TVirtualRWMutex.h"
+#include "ROOT/TReentrantRWLock.hxx"
+#include "ROOT/TRWSpinLock.hxx"
+
+#include "../src/TRWMutexImp.h"
```

Comment:
Why? Isn't it obvious.  I would just add a comment that says "This unit test tests a private class directly", isn't that what 'unit' test are for?